src/fsharp/CompileOptions.fs

```diff
@@ -172,6 +172,32 @@ let DumpCompilerOptionBlocks blocks = List.iter dumpCompilerOptionBlock blocks
 let isSlashOpt (opt:string) = 
     opt.[0] = '/' && (opt.Length = 1 || not (opt.[1..].Contains "/"))
 
+module ResponseFile =
+
+    type ResponseFileData = ResponseFileLine list
+    and ResponseFileLine =
+        | CompilerOptionSpec of string
+        | Comment of string
+
+    let parseFile path : Choice<ResponseFileData,Exception> =
+        let parseLine (l: string) =
+            match l with
+            | s when String.IsNullOrWhiteSpace(s) -> None
+            | s when l.StartsWith("#") -> Some (ResponseFileLine.Comment (s.TrimStart('#')))
+            | s -> Some (ResponseFileLine.CompilerOptionSpec (s.Trim()))
+
+        try
+            use stream = FileSystem.FileStreamReadShim path
+            use reader = new System.IO.StreamReader(stream, System.Text.Encoding.UTF8)
```

Comment:
My concern was more about the creation options of `StreamReader`.
You instantiate it with UTF8 encoding, in contrast roslyn doesn't provide an encoding at all, but enables the `detectEncodingFromByteOrderMarks`.